web/compose-compiler-integration/build.gradle.kts

```diff
@@ -0,0 +1,134 @@
+fun cloneTemplate(templateName: String, contentMain: String, contentLib: String): File {
+    val tempDir = file("${project.buildDir.absolutePath}/temp/cloned-$templateName")
+    tempDir.deleteRecursively()
+    tempDir.mkdirs()
+    file("${projectDir.absolutePath}/main-template").copyRecursively(tempDir)
+    // tempDir.deleteOnExit()
+    File("$tempDir/src/commonMain/kotlin/Main.kt").printWriter().use { out ->
+        out.println(contentMain)
+    }
+    File("$tempDir/lib/src/commonMain/kotlin/Lib.kt").printWriter().use { out ->
+        out.println(contentLib)
+    }
+    return tempDir
+}
+
+fun build(
+    caseName: String,
+    directory: File,
+    failureExpected: Boolean = false,
+    vararg buildCmd: String = arrayOf("build", "jsNodeRun")
+) {
+    val isWin = System.getProperty("os.name").startsWith("Win")
+    val procBuilder = if (isWin) {
+        ProcessBuilder("gradlew.bat", *buildCmd)
+    } else {
+        ProcessBuilder("bash", "./gradlew", *buildCmd)
+    }
+    val proc = procBuilder
+        .directory(directory)
+        .redirectOutput(ProcessBuilder.Redirect.PIPE)
+        .redirectError(ProcessBuilder.Redirect.PIPE)
+        .start()
+
+    proc.waitFor(5, TimeUnit.MINUTES)
```

Comment:
why we need this?

Reply:
`proc.waitFor(5, TimeUnit.MINUTES)`
Causes the current thread to wait, if necessary, until the subprocess represented by this Process object has terminated, or the specified waiting time elapses.

We can poll the `exitValue` value or `isAlive()` in the loop instead. But anyway we need to set some timeout. So `waitFor` seems to be the most concise solution for the task.